### PR TITLE
Added OAS2/3 support

### PR DESCRIPTION
**Pull request by ruks**
Added OAS2/3 support

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+        } else {
+            Swagger swagger = parseAttemptForV2.getSwagger();
+            Info info = swagger.getInfo();
+            OASParserUtil.updateValidationResponseAsSuccess(validationResponse, apiDefinition, swagger.getSwagger(),
+                    info.getTitle(), info.getVersion(), swagger.getBasePath(), info.getDescription());
+            validationResponse.setParser(this);
+            if (returnJsonContent) {
+                validationResponse.setJsonContent(OASParserUtil.getSwaggerJsonString(parseAttemptForV2.getSwagger()));
+            }
+
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java]: Additional new lines

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 2]
```diff
@@ -284,6 +284,10 @@ paths:
           required: true
           schema:
             $ref: '#/definitions/API'
+        - in: query
+          name: oas_version
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we use 'oasVersion' or 'openAPIVersion'? We usually don't use _s in parameters.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/definitions/OAS3ParserTest.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.api.model.URITemplate;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.Assert.*;
```
Comment (hasuniea) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/definitions/OAS3ParserTest.java]: import only the required packages.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java (hunk 1 above)
Comment (ruks) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java]: fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 2 above)
Comment (ruks) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/definitions/OAS3ParserTest.java (hunk 3 above)
Comment (ruks) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/definitions/OAS3ParserTest.java]: fixed

**Pull request merged**